lib/3scale/backend/transactor.rb

```diff
@@ -58,34 +58,23 @@ def validate(oauth, provider_key, report_usage, params, extensions)
         # service_id cannot be taken from params since it might be missing there
         service_id = service.id
 
-        app_id, user_id = params[:app_id], params[:user_id]
+        app_id = params[:app_id]
         # TODO: make sure params are nil if they are empty up the call stack
         # Note: app_key is an exception, as it being empty is semantically
         # significant.
         params[:app_id] = nil if app_id && app_id.empty?
-        params[:user_id] = nil if user_id && user_id.empty?
-
-        # Now OAuth tokens also identify users, so must check tokens anyway if
-        # at least one of app or user ids is missing.
-        #
-        # NB: so what happens if we call an OAuth method with user_key=K and
-        # user_id=U? It is effectively as if app_id was given and no token would
-        # need to be checked, but we do... That is not consistent. And madness.
-        # Each time I try to understand this I feel I'm becoming dumber...
```

Comment:
A piece of wisdom unfortunately gone forever.

Reply:
It will always be in the git history :)